chainer/gradient_check.py

```diff
@@ -279,29 +284,47 @@ def check_backward(func, x_data, y_grad, params=(),
     params_grad = [param.grad for param in params]
 
     xp = cuda.get_array_module(*xs)
-    one = xp.array(1., dtype)
+    xs_directions = [
+        xp.random.normal(size=x.shape).astype(dtype) for x in casted_xs]
+    param_directions = [
+        xp.random.normal(size=x.shape).astype(dtype) for x in param_data]
```

Comment:
Please care the case `dtype == None`.

Reply:
When `dtype==None`, `astype` returns itself. So, we don't need to change this code, right?

Reply:
Sorry, in such case, numpy uses float64. I'll fix it.